subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/BuildFailureListener.kt

```diff
@@ -0,0 +1,73 @@
+package com.avito.android.build_verdict
+
+import com.avito.utils.logging.CILogger
+import com.google.gson.Gson
+import org.gradle.BuildResult
+import org.gradle.api.Action
+import org.gradle.api.execution.TaskExecutionGraph
+import org.gradle.api.file.Directory
+import java.io.File
+import java.io.PrintWriter
+import java.io.StringWriter
+
+internal class BuildFailureListener(
+    private val graph: TaskExecutionGraph,
+    private val buildVerdictDir: Directory,
+    private val logs: Map<TaskPath, StringBuilder>,
+    private val ciLogger: CILogger,
+    private val gson: Gson
+) : Action<BuildResult> {
+
+    override fun execute(result: BuildResult) {
+        result.failure?.apply {
+            onFailure(this)
+        }
+    }
+
+    private fun onFailure(failure: Throwable) {
+        val failedTasks = graph.allTasks
+            .filter { it.state.failure != null }
+        val verdict = gson.toJson(
```

Comment:
i want to see `Writer` extracted as separate abstraction;
this way `private val gson: Gson` should not be in `BuildFailureListener` constructor (it seemed weird for me as start, and because of hardcoded writer here)